main/src/cgeo/geocaching/calculator/CoordinatesCalculateUtils.java

```diff
@@ -0,0 +1,163 @@
+
+        // Add all the left over equations to the variable bank.
+        for (final VariableData var : variables) {
+            varBank.add(var);
+        }
+
+        return returnList;
+    }
+
+    /** Creates a calc-state out of given lat- and lon-formula and list of variables
+     *
+     * @param latText formula / coordinates for latitude
+     * @param lonText formula / coordinates for longitude
+     * @param variableDataList list with already known variables / equations for formula
+     *
+     * @return calc state.
+     * */
+    public static CalcState createCalcState(final String latText, final String lonText, final List<VariableData> variableDataList) {
+
+        String coordinateChars = ""; // All the characters that appear in the coordinate representation.
+
+        char latHem = 'N';
+        if (latText.length() > 0) {
+            final char first = latText.charAt(0);
+            if (first == 'N' || first == 'S') {
+                latHem = first;
+                coordinateChars = coordinateChars.concat(latText.substring(1));
+            } else {
+                coordinateChars = coordinateChars.concat(latText);
+            }
+        }
+
+        char lonHem = 'W';
+        if (lonText.length() > 0) {
+            final char first = lonText.charAt(0);
+            if (first == 'E' || first == 'W' || first == 'O') {
+                lonHem = first;
+                coordinateChars = coordinateChars.concat(lonText.substring(1));
+            } else {
+                coordinateChars = coordinateChars.concat(lonText);
+            }
+        }
+
+        List<VariableData> equData = new ArrayList<>();
+        equData = CoordinatesCalculateUtils.sortVariables(equData, variableDataList, coordinateChars, new CaseCheck(true));
+
+        String equationStrings = "";
+        for (final VariableData equ : equData) {
+            equationStrings = equationStrings.concat(equ.getExpression());
+        }
+
+        // replace the old free variables list with a newly created ones.
+        List<VariableData> freeVarData = new ArrayList<>();
+        freeVarData = CoordinatesCalculateUtils.sortVariables(freeVarData, variableDataList, equationStrings, new CaseCheck(false));
+
+        final List<ButtonData> butData = new ArrayList<>();
+
+        final CalcState calcState = new CalcState(Settings.CoordInputFormatEnum.Plain,
+            latText,
+            lonText,
+            latHem,
+            lonHem,
+            butData,
+            equData,
+            freeVarData,
+            variableDataList);
+
+        return calcState;
+    }
+
+    /**
+     * Find if a variable exists in the supplied list with the given name
+     *
+     * @param name name to search for
+     * @param list list of variables
+     * @return first occurrence of the variable if it can found, 'null' otherwise
+     */
+    private static VariableData getVariable(final char name, final List<VariableData> list, final boolean remove) {
```

Comment:
This methods performance would profit from "list" being a map (char -> VariableData).  Would that be an adequate data structure for a collection of VariableData's?